### PR TITLE
Integrate config and session property for history matching threshold

### PR DESCRIPTION
**Pull request by hantangwangd**
## Description

This PR integrate config property into session property for history-input-table-statistics-matching-threshold, so that we can get the property value uniformly through method `getHistoryInputTableStatisticsMatchingThreshold()` in `SystemSessionProperties`. 

Besides, there exists a problem in the previous implementation logic when we trying to explicitly set the threshold to `0.0` through session property. By setting the value to 0.0 through session property, we intended to indicate that we will use historical statistics only when the input tables have exactly the same stats with those in the historical stats, but that's not the case in previous implement logic as we would then get the value from configuration property in this scenario, which is default not 0.0. This PR fix the problem as well.

## Test Plan

 - Newly added test cases in `TestHiveHistoryBasedStatsTracking` including the scenario described above

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by jaystarshot — APPROVED** [review]

**Comment by steveburnett** [comment]
Is any documentation update needed for this?

**Comment by hantangwangd** [comment]
@steveburnett Thanks for the reminder, I have update the description for `history_input_table_statistics_matching_threshold` in docs `history-based-optimization.rst` to amend the description of default value and the behavior when set to `0.0`, please take a look. Additionally, should we open a dedicate issue for fixing the description of other session properties in `history-based-optimization.rst`? The description for these session properties should combine with the corresponding config properties.

cc @feilong-liu @jaystarshot

**Comment by steveburnett** [comment]
> @steveburnett Thanks for the reminder, I have updated the description for `history_input_table_statistics_matching_threshold` in docs `history-based-optimization.rst` to amend the description of default value and the behavior when set to `0.0`, please take a look. Additionally, should we open a dedicate issue for fixing the description of other session properties in `history-based-optimization.rst`? The description for these session properties should combine with the corresponding config properties.
> 
> cc @feilong-liu @jaystarshot

If you find inconsistencies or gaps in the doc for the HBO session properties, I would welcome updates. @feilong-liu @jaystarshot, what do you think?

File: presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst [hunk 1]
```diff
@@ -24,16 +24,17 @@ How to use HBO
...
 restrict_history_based_optimization_to_complex_query        Enable history based optimization only for complex queries, i.e. queries with join and aggregation                                                                                                          True
-history_input_table_statistics_matching_threshold           When the size difference between current table and history table exceed this threshold, do not match history statistics. When value is 0, use the default value set by hbo.history-matching-threshold       0
-=========================================================== =========================================================================================================================================================================================================== ===============
+history_input_table_statistics_matching_threshold           When the size difference between current table and history table exceed this threshold, do not match history statistics. When value is 0.0, only match history statistics when the size of the two          ``hbo.history-matching-threshold``
+                                                            are exactly the same
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst:36]:
```suggestion
                                                            are exactly the same.
```
A very tiny nit, for punctuation.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the doc update! A very tiny nit, for punctuation.

File: presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst (hunk 1 above)
Comment (hantangwangd) [reply presto-docs/src/main/sphinx/optimizer/history-based-optimization.rst]: Thanks for the suggestion. I find the other session property descriptions all do not have the last punctuation, so add it for  all of them. Please take a look when available.

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs)
Pulled updated branch, new docs build, looks good. 
Thanks!

**Review by tdcmeehan — COMMENTED** [review]
LGTM.  @feilong-liu @jaystarshot can you also take a look?

**Review by feilong-liu — APPROVED** [review]
Just to double confirm, the default behavior will not change, i.e. the default match threshold stays the same?

**Comment by hantangwangd** [comment]
> Just to double confirm, the default behavior will not change, i.e. the default match threshold stays the same?

Yes, the default behavior will not change, the actual default value before and after this change are all `0.1`.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**